pkg/controller/direct/iam/partialpolicy_controller.go

```diff
@@ -0,0 +1,403 @@
+
+	// Update KRM status
+	newKRMStatus := newiamv1beta1.IAMPartialPolicyStatus{
+		ObservedGeneration:  a.desired.GetGeneration(),
+		LastAppliedBindings: desiredPartialPolicyWithStatus.Status.LastAppliedBindings,
+		AllBindings:         desiredPartialPolicyWithStatus.Status.AllBindings,
+	}
+	readyCondition := k8s.NewCustomReadyCondition(corev1.ConditionTrue, k8s.UpToDate, k8s.UpToDateMessage)
+	if err := createOp.UpdateStatus(ctx, &newKRMStatus, &readyCondition); err != nil {
+		return fmt.Errorf("updating status for %v: %w", k8s.GetNamespacedName(a.desired), err)
+	}
+	return nil
+}
+
+func (a *IAMPartialPolicyAdapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := getLogger(ctx)
+	log.V(2).Info("updating/applying IAMPartialPolicy", "name", k8s.GetNamespacedName(a.desired))
+	mapCtx := &direct.MapContext{}
+
+	if a.actualReferencedResourcePolicy == nil {
+		return fmt.Errorf("no IAM policy found for referenced resource %s during update for IAMPartialPolicy %v; requeueing for Find", a.desired.Spec.ResourceReference.GetNamespacedName(), k8s.GetNamespacedName(a.desired))
+	}
+
+	livePolicyForMerge := &newiamv1beta1.IAMPolicy{}
+	livePolicyForMerge.Spec = *IAMPolicySpec_FromProto(mapCtx, a.actualReferencedResourcePolicy)
+
+	resolver := IAMMemberIdentityResolver{Iamclient: a.iamClient, Ctx: ctx}
+
+	desiredPartialPolicyWithStatus, err := ComputePartialPolicyWithMergedBindings(a.desired, livePolicyForMerge, resolver)
+	if err != nil {
+		return fmt.Errorf("computing partial policy for update for %v: %w", k8s.GetNamespacedName(a.desired), err)
+	}
+
+	finalPolicyToSet := toDesiredPolicy(desiredPartialPolicyWithStatus, livePolicyForMerge)
+
+	// todo acpana better comparison, etag?
+	gcpUpdateNeeded := !reflect.DeepEqual(livePolicyForMerge.Spec.Bindings, finalPolicyToSet.Spec.Bindings) ||
+		!reflect.DeepEqual(livePolicyForMerge.Spec.AuditConfigs, finalPolicyToSet.Spec.AuditConfigs)
+
+	if gcpUpdateNeeded {
+		log.V(2).Info("GCP IAM policy change detected, calling SetPolicy", "name", k8s.GetNamespacedName(a.desired))
+		_, err = a.iamClient.SetPolicy(ctx, finalPolicyToSet)
+		if err != nil {
+			return fmt.Errorf("setting IAM policy for %v: %w", k8s.GetNamespacedName(a.desired), err)
+		}
+		log.V(2).Info("successfully applied IAMPartialPolicy for update", "name", k8s.GetNamespacedName(a.desired))
+	} else {
+		log.V(2).Info("no change in GCP IAM policy needed", "name", k8s.GetNamespacedName(a.desired))
+	}
+
+	// Always update KRM status to reflect observed generation and latest computed bindings,
+	// even if GCP didn't change, as spec or other KRM details might have.
+	newKRMStatus := newiamv1beta1.IAMPartialPolicyStatus{
+		ObservedGeneration:  a.desired.GetGeneration(),
+		LastAppliedBindings: desiredPartialPolicyWithStatus.Status.LastAppliedBindings,
+		AllBindings:         desiredPartialPolicyWithStatus.Status.AllBindings,
```

Comment:
Just curious, is it the existing behavior? I think the IAM policy can have up to 1500 bindings, so the status could be hard to read sometimes. I wondered if we want to filter the `status` by member or etc when the size is large, or making this field filterable by user configuration.